### PR TITLE
Test framework -  upload test

### PR DESCRIPTION
**Pull request by jeremygoh**
@david-yz-liu @WilsonChiang 
Builds on top of #2321 which is to be merged in and fixes the bug with polymorphic urls.

This PR allows us to upload a test file for a given assignment.

Cases around test file uploading still to be considered:
- replacing a test file
- uploading multiple test file (or changing the form such that it only allows uploading of one test file)
- modifying the form to render already uploaded files properly

### CODE REVIEW

**Comment by david-yz-liu** [comment]
@jeremygoh @WilsonChiang One of you should fix the Hound comments here.

**Comment by WilsonChiang** [comment]
@jeremygoh I'm fixing the hound comments currently.

**Comment by WilsonChiang** [comment]
@jeremygoh Hmm.. should I be able to push to your branch? Seems like I don't have permission.

**Comment by david-yz-liu** [comment]
@WilsonChiang You can't push to each other's forks unless you set the permissions for it. You can always make a pull request into Jeremy's branch.

**Comment by jeremygoh** [comment]
@david-yz-liu a lot of the code that Hound has commented on, and also the block you mentioned, are pieces of code from the original test-framework branch. Should I make the suggested modifications in this PR or in another?

File: app/helpers/automated_tests_helper.rb [hunk 1]
```diff
@@ -85,61 +113,117 @@ def create_ant_test_files(assignment)
...
 
-    # Filter out files that need to be created and updated:
-    testfiles.each_key do |key|
+    # Retrieve all test support files
+    testsupporters = params[:test_support_files_attributes]
+
+    # Now check for duplicate file names in test support files
+    unless testsupporters.nil?
+      testsupporters.values.each do |tfile|
+        if tfile['file_name'].respond_to?(:original_filename)
```
Comment (WilsonChiang) [inline app/helpers/automated_tests_helper.rb]: @david-yz-liu These seem like false positives?

**Pull request merged**